packages/teams-js/src/internal/communication.ts

```diff
@@ -267,21 +267,28 @@ function sendMessageToParentHelper(actionName: string, args: any[]): MessageRequ
   return request;
 }
 
+const processMessageLogger = communicationLogger.extend('processMessage');
+
 /**
  * @internal
  * Limited to Microsoft-internal use
  */
 function processMessage(evt: DOMMessageEvent): void {
   // Process only if we received a valid message
   if (!evt || !evt.data || typeof evt.data !== 'object') {
+    processMessageLogger('Unrecognized message format received by app, message being ignored');
     return;
   }
 
   // Process only if the message is coming from a different window and a valid origin
-  // valid origins are either a pre-known
+  // valid origins are either a pre-known origin or one specified by the app developer
+  // in their call to app.initialize
   const messageSource = evt.source || (evt.originalEvent && evt.originalEvent.source);
   const messageOrigin = evt.origin || (evt.originalEvent && evt.originalEvent.origin);
   if (!shouldProcessMessage(messageSource, messageOrigin)) {
+    processMessageLogger(
+      'Message being ignored by app because it is not coming from a different window with a valid origin',
```

Comment:
>not coming from a different window with a valid origin [](http://example.com/codeflow?start=50&length=54)

For some reason I had a hard time parsing this message. Is this the intent:
"Message being ignore by app because it is either coming from this own window or it's coming from a different window but with an invalid origin" ?

Reply:
Yes, will update the language